Shopping Cart, Iugu Payment & more...

patrickemuller opened the pull request:
- Shopping cart
- Iugu payment with credit card & bank slip
- Purchase Mailers
- Purchase updates from Iugu (webhooks)
- Validations for credit card form
  ...and more

patrickemuller committed:
Adds devise to present users & create tasks to migrate them

patrickemuller committed:
Adds shopping cart

patrickemuller committed:
Migrate purchases to old_purchases

This moves all actual purchases to old_purchases so this way the new payment gateway can use the Purchases Table to store the new credit card purchases (IUGU)

patrickemuller committed:
Adds Purchase & Order Models and relate with users

patrickemuller committed:
Adds checkout page (without form) and fixes redirects after login

patrickemuller committed:
Adds iugu & secure token gems

patrickemuller committed:
Adds missed option to user return to payment after sign up

patrickemuller committed:
Adds payment fields to purchase model

patrickemuller committed:
Moves PurchaseStatus to OldPurchaseStatus to avoid conflicts

patrickemuller committed:
Adds rough form to pay purchases

patrickemuller committed:
Fixes missed call of local registrations controller on routes.rb

patrickemuller committed:
Fixes visualization of remaining quantity of an offer

patrickemuller committed:
Fixes problems when adding more offer than have on stock

patrickemuller committed:
Adds webhooks from Iugu to update purchase's status

patrickemuller committed:
Moves checkouts#update to purchases#update

- Improves seeds.rb
- Creates method to calculate total of order
- Creates method to calcular total of purchase

patrickemuller committed:
Fixes problems with messages with charge.message on checkout

patrickemuller committed:
Changes redirect to cart when add/remove offer from cart

patrickemuller committed:
Adds option to accept terms of buy on checkout page

patrickemuller committed:
Moves add to cart action to "Comprar" button on offers#show

patrickemuller committed:
Adds basic layout to cart page

patrickemuller committed:
Adds basic layout to checkout page

patrickemuller committed:
Adds basic layout to checkout success page

patrickemuller committed:
Adds mailers for pending/confirmed payments

patrickemuller committed:
Fixes problems with admin/user

patrickemuller committed:
Fixes problems with purchase mailer pending_payment.html.slim

ssjr commented on Gemfile:
```diff
@@ -39,6 +39,8 @@ gem 'enumerate_it'
 # For masked fields
 gem 'maskedinput-rails'
+# Payment gateway
+gem 'iugu'
```
quando é pagamento eu gosto de deixar explícito a versão da gem

ssjr commented on app/admin/order.rb:
```diff
@@ -0,0 +1,3 @@
+# ActiveAdmin.register Order do
```
certo?

ssjr commented on app/assets/javascripts/tribo_viva/checkouts/checkout.js.coffee:
```diff
@@ -0,0 +1,97 @@
+TriboViva.Checkouts.Checkout =
+  init: ->
+    Iugu.setAccountID("79d7660c-6169-4e07-aed9-8a0883d5249a")
+    Iugu.setTestMode(true) # disable on production
```
essas duas linhas tu poderia colocar numa `<div>` com um ID, tipo o que acontece no [initjs](https://github.com/josemarluedke/initjs) que tem uma div com data attributes, tipo

``` slim
#iugu-widget[data-account="#{Rails.application.secrets.iugu_account_id}" data-environment="#{Rails.env}"]
```

e no js

``` coffee
widget = $('#iugu-widget')
Iugu.setAccountID(widget.data('account'))
Iugu.setTestMode(widget.data('environment') != 'production')
```

ssjr replied:
e não tem que dar o `Iugu.setup()`?

patrickemuller replied:
I think that setup is just called when you're using Formatter.js

ssjr commented on app/assets/javascripts/tribo_viva/checkouts/checkout.js.coffee:
```diff
@@ -0,0 +1,97 @@
+    Iugu.setTestMode(true) # disable on production
+
+    # Set mask for card expiration date
+    $(".js-credit-card-number").mask("9999 9999 9999 9999")
```
eu tiraria o jquery mask que tem hoje no app e usaria o [Formatter.js](https://github.com/firstopinion/formatter.js) em todo ele (não só no pagamento). Se tu for ver o código do [iugu.js](https://js.iugu.com/v2) descompactado, vai ver que ele vê se tem o formatter e se tiver, já faz toda a mão das máscaras automaticamente de acordo com o cartão que tu digita. colocando uma classe da bandeira no form e ajustando automaticamente a máscara pro tipo do cartão. visa e master tem 16 dígitos, funcionaria com essa máscara de hoje. mas amex por exemplo tem 15 dígitos, o mask fixo desse jeito nunca ia deixar o usuário digitar o cartão

patrickemuller replied:
Do you have the link to `Formatter.js` gem? I didn't found.

ssjr replied:
não tem gem, só o js. e tem o https://github.com/firstopinion/formatter.js/blob/master/dist/jquery.formatter.js e o https://github.com/firstopinion/formatter.js/blob/master/dist/formatter.js no caso do iugu é o sem jquery. se for usar no app com jquery teria que baixar os dois (mas não tem pq)

ssjr commented on app/assets/javascripts/tribo_viva/checkouts/checkout.js.coffee:
```diff
@@ -0,0 +1,97 @@
+
+    # Set mask for card expiration date
+    $(".js-credit-card-number").mask("9999 9999 9999 9999")
+    $(".js-credit-card-card-expiration").mask("99/99")
```
aqui também, se tem o formatter o iugu bota máscara sozinho

ssjr commented on app/assets/javascripts/tribo_viva/checkouts/checkout.js.coffee:
```diff
@@ -0,0 +1,97 @@
+
+      if creditCardCheckBox.attr('checked')
+        # first of all, handle errors on credit card form
+        if !Iugu.utils.validateCreditCardNumber(cc, brand)
```
pq valida o cartão só quando o usuário envia o form e não num `$('.js-credit-card-number').on('change')` da vida? (outros abaixo também)

patrickemuller replied:
You think that is better to validate every time it changes? @filipelinhares what you say? Validate on form submit or when fields change?

filipelinhares replied:
I think when fields change. It's better for the user.

ssjr commented on app/assets/javascripts/tribo_viva/checkouts/checkout.js.coffee:
```diff
@@ -0,0 +1,97 @@
+              e.preventDefault()
+            else
+              $('#token').val(data.id)
+              console.log $('#token').val()
```
log

ssjr commented on app/assets/javascripts/tribo_viva/checkouts/checkout.js.coffee:
```diff
@@ -0,0 +1,97 @@
+            else
+              $('#token').val(data.id)
+              console.log $('#token').val()
+              console.log form.get(0)
```
log

ssjr commented on app/assets/stylesheets/organisms/_payment_form.scss:
```diff
@@ -0,0 +1,176 @@
+}
+/* Do not forget to store your images in a secure server */
+#js-creditcard-form .input-group .icon.ccic-name {
+    background: url("http://storage.pupui.com.br/9CA0F40E971643D1B7C8DE46BBC18396/assets/ccic-name.1cafa1882fdd56f8425de54a5a5bbd1e.png") no-repeat;
```
externo?

patrickemuller replied:
This SCSS is from Iugu. Probably this will change when @cbgabe or @filipelinhares put some more design.

ssjr commented on app/assets/stylesheets/organisms/_payment_form.scss:
```diff
@@ -0,0 +1,176 @@
+    background: url("http://storage.pupui.com.br/9CA0F40E971643D1B7C8DE46BBC18396/assets/ccic-name.1cafa1882fdd56f8425de54a5a5bbd1e.png") no-repeat;
+}
+#js-creditcard-form .input-group .icon.ccic-exp {
+    background: url("http://storage.pupui.com.br/9CA0F40E971643D1B7C8DE46BBC18396/assets/ccic-exp.05e708b1489d5e00c871f20ba33bbff3.png") no-repeat;
```
externo?

ssjr commented on app/assets/stylesheets/organisms/_payment_form.scss:
```diff
@@ -0,0 +1,176 @@
+    background: url("http://storage.pupui.com.br/9CA0F40E971643D1B7C8DE46BBC18396/assets/ccic-exp.05e708b1489d5e00c871f20ba33bbff3.png") no-repeat;
+}
+#js-creditcard-form .input-group .icon.ccic-brand {
+    background: url("http://storage.pupui.com.br/9CA0F40E971643D1B7C8DE46BBC18396/assets/ccic-brands.48dba03883007f86e118f683dcfc4297.png") no-repeat;
```
externo?

ssjr commented on app/assets/stylesheets/organisms/_payment_form.scss:
```diff
@@ -0,0 +1,176 @@
+    background: url("http://storage.pupui.com.br/9CA0F40E971643D1B7C8DE46BBC18396/assets/ccic-brands.48dba03883007f86e118f683dcfc4297.png") no-repeat;
+}
+#js-creditcard-form .input-group .icon.ccic-cvv {
+    background: url("http://storage.pupui.com.br/9CA0F40E971643D1B7C8DE46BBC18396/assets/ccic-cvv.1fe78dcc390427094bdc14dedea10f34.png") no-repeat;
```
externo?

ssjr commented on app/controllers/checkouts_controller.rb:
```diff
@@ -0,0 +1,95 @@
+class CheckoutsController < ApplicationController
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized, except: :update
```
aqui e embaixo não tem action update no controller, não entendi o except

patrickemuller replied:
Will remove this. This is because I moved the `update` action to `PurchasesController` early.

ssjr commented on app/controllers/checkouts_controller.rb:
```diff
@@ -0,0 +1,95 @@
+    authorize :checkout
+
+    # get payment method, if blank, it's credit card, if not, it's bank_slip
+    param          = (params[:token].empty? ? "method" : "token").to_sym
```
pq não direto?

``` ruby
param = params[:token].empty? ? :method : :token
```

patrickemuller replied:
Good idea :D

ssjr commented on app/controllers/checkouts_controller.rb:
```diff
@@ -0,0 +1,95 @@
+
+    # get payment method, if blank, it's credit card, if not, it's bank_slip
+    param          = (params[:token].empty? ? "method" : "token").to_sym
+    payment_method = params[:method] == '' ? 'credit_card' : 'bank_slip'
```
pq não coloca direto o input method == credit_card no form, como já acontece com o bank_slip?

patrickemuller replied:
When credit card is select on Iugu form, the "method" attribute is empty, but when bank_slip so this have an value. Don't ask why they did this way...

ssjr commented on app/controllers/checkouts_controller.rb:
```diff
@@ -0,0 +1,95 @@
+    # Add all current items from cart to orders and purchase
+    session[:shopping_cart].each do |item|
+      offer = Offer.find(item['id'])
+      order = Order.create(offer: offer,
```
não tá usando `order` em nenhum lugar, não precisaria da variável

ssjr commented on app/controllers/checkouts_controller.rb:
```diff
@@ -0,0 +1,95 @@
+    charge = Iugu::Charge.create({
+      # here is where the method is determined, :token for CC or :method bank_slip
+      param => params[param],
+      email: current_user.email,
```
ainda acho que devia dar opção pro usuário escolher esses campos (email e os do `payer`) no formulário de compra, pode ser outra pessoa a dona do cartão (poderia vir pré-preenchido, mas com opção do cara botar outros valores)

patrickemuller replied:
Yes, this can be changed, but for now I'll let the user just fill the informations about the card, once that we don't need to confirm CPF or Payer name to make the purchase.

ssjr commented on app/controllers/checkouts_controller.rb:
```diff
@@ -0,0 +1,95 @@
+  end
+
+  def success
+    @purchase = Purchase.find_by(invoice_id: params[:invoice_id])
```
aqui não seria o caso de dar 404 se não achar? `Purchase.find_by!`

ssjr commented on app/controllers/checkouts_controller.rb:
```diff
@@ -0,0 +1,95 @@
+  end
+
+  def store_location
+    store_location_for(:user, request.path)
```
isso vai em todas as requisições, mesmo em post/patch/update/etc. depois ele tentaria redirecionar pra essas locations mas provavelmente não vai ter um path com a mesma url do post/etc e vai dar erro. teria que ser só pra requisições `GET`

ssjr commented on app/controllers/purchases_controller.rb:
```diff
@@ -1,32 +1,31 @@
-    if @purchase.status != PurchaseStatus::CONFIRMED
-      flash[:alert] = 'Você deve fazer o upload do recibo de pagamento!'
-      redirect_to purchase_path(@purchase)
+    if purchase.exists?
```
com um `purchase = Purchase.find_by(invoice_id: data[:id])` poderia ser só `if purchase`

ssjr commented on app/controllers/purchases_controller.rb:
```diff
@@ -1,32 +1,31 @@
+    if purchase.exists?
+      if event == 'invoice.refund' || event == 'invoice.status_changed'
+        PurchaseMailer.confirmed_payment(purchase).deliver_now if data[:status] == 'paid'
+        purchase.first.update_attributes(status: data[:status])
```
e com o `find_by` no lugar do `where`, sairia esse `.first`

ssjr commented on app/controllers/purchases_controller.rb:
```diff
@@ -1,32 +1,31 @@
+        purchase.first.update_attributes(status: data[:status])
+        render nothing: true, status: :ok, content_type: "text/html"
+      elsif event == 'invoice.payment_failed'
+        purchase.first.update_attributes(status: PurchaseStatus::CANCELED)
```
aqui também

ssjr commented on app/controllers/users/registrations_controller.rb:
```diff
@@ -0,0 +1,7 @@
+  protected
+
+  def after_sign_up_path_for(resource)
+    stored_location_for(:user) || super
```
acessou o carrinho uma vez = sempre vai voltar pra algo do `CheckoutsController` depois de fazer o login. teria que armazenar pra todo app, no `ApplicationController`

ssjr commented on app/models/offer.rb:
```diff
@@ -13,11 +15,18 @@ class Offer < ActiveRecord::Base
+    stock - Purchase.joins(:orders)
+                    .where(status: PurchaseStatus::PAID,
+                           orders:{offer_id: self.id})
+                    .sum(:quantity)
```
"tudo isso -1" pq o coordenador de entrega sempre tem a dele

patrickemuller replied:
Sure?

ssjr replied:
já li 2x, mas achei só nesse print
<img width="779" alt="screen shot 2015-07-16 at 10 07 06" src="https://cloud.githubusercontent.com/assets/803887/8723334/3789a3ae-2ba3-11e5-82eb-5b5062dd0379.png">

ssjr commented on app/models/order.rb:
```diff
@@ -0,0 +1,20 @@
+  belongs_to :offer
+  belongs_to :purchase
+
+  validates :offer_id, :quantity, :purchase_id, :offer_value, presence: true
```
``` ruby
validates :offer
```

ssjr commented on app/models/order.rb:
```diff
@@ -0,0 +1,20 @@
+
+  protected
+
+  def update_purchase_value
```
isso aqui ficou bem confuso. pq não dá um update no controller já que tu só cria uma vez por lá e tem todos os orders por lá também?

ssjr commented on app/models/purchase.rb:
```diff
@@ -1,43 +1,13 @@
+  has_many :orders, dependent: :destroy
 
-  has_enumeration_for :status, with: PurchaseStatus, create_helpers: true
+  validates :user_id, :status, :total, presence: true
```
``` ruby
validates :user
```

ssjr commented on app/views/checkouts/checkout.html.slim:
```diff
@@ -0,0 +1,74 @@
+                h3 1. Selecione a opção de pagamento
+                .btn-group.payment-method-btn-group data-toggle="buttons"
+                  label#js-credit-card-checkbox-label.btn.btn-default.active Cartão de crédito
+                  = check_box_tag :method, '', true, id: 'js-payment-method-credit-card', style: 'display: none;'
```
aqui que eu falei lá no js, de colocar `credit_card` como value

patrickemuller replied:
Basically because for Iugu when you pass a method value this must be `bank_slip`, and when you pass a token this must be `credit_card`.

ssjr commented on app/views/user_mailer/massive_reset_password.html.slim:
```diff
@@ -0,0 +1,12 @@
+h1 Olá #{@user.name}
+
+p
+  | Devido a uma manutenção recente no servidor, o qual trará melhoras significativas na forma
```
ficou meio ~bad~ o texto. tem que trazer mais empolgação pro usuário. algo tipo

> Olá fulado.
> Você já se cadastrou na rede TriboViva e está recebendo esse email pois temos ótimas notícias!
> Estamos evoluindo com a plataforma e agora todos os usuários terão uma senha e poderão comprar com cartão de crédito entre outras novidades.
> Estamos melhorando a plataforma um pouco de cada vez, sempre buscando o melhor para você. Anteriormente você não precisava de uma senha. Agora será necessária e para não ter que preencher todos aqueles campos novamente, optamos por enviar um email para resetar sua senha. Clique aqui para trocar sua senha, e se por algum motivo não conseguir trocar sua senha, vá novamente no site, na parte de login e clique em "Esqueci minha senha", digite seu email que outro email será enviado automaticamente.
> Estamos animados com esse grande passo, e é só o começo
> Um abraço da tribo

ou algo nesse sentido. o marcos deve ter uma ideia boa de texto

patrickemuller replied:
@cbgabe we need to discuss about the content of this mailers.

ssjr commented on config/secrets.yml:
```diff
@@ -12,6 +12,7 @@
 
 development:
   secret_key_base: 12ebb4f25d25b92275e08be8e4f3c2a2e206b393315fbbd6fd8041bef769a4415830cac19803b2bae0c10d0da749fcc9f4e578ef64ac52522639831a019d6523
+  iugu_api_key: <%= ENV["IUGU_API_KEY"] %>
```
na vdd é só chave de teste, teoricamente não teria problema em colocar ela direto aqui em dev

ssjr commented on db/migrate/20150707175305_add_devise_to_users.rb:
```diff
@@ -0,0 +1,49 @@
+  def self.down
+    # By default, we don't want to make any assumption about how to roll back a migration when your
+    # model already existed. Please edit below which fields you would like to remove in this migration.
+    raise ActiveRecord::IrreversibleMigration
```
na vdd pode ter rollback sim pq é só uns campo a mais né

ssjr commented on db/seeds.rb:
```diff
@@ -60,9 +61,13 @@
+    5.times do |index2|
+      user  = User.find(index2+1)
+      offer = Offer.last
+      OldPurchase.create(user: user, offer: offer, amount: 1, status: OldPurchaseStatus::CONFIRMED,
```
não precisa mais do `OldPurchase` no seed...

patrickemuller replied:
I'll remove everything related to `OldPurchase` when we define the final scope for this new payment gateway implementation. But probably I'll remove this after the merge, making another PR.

ssjr commented:
https://travis-ci.org/codelandev/tribo-viva/pull_requests & https://codeclimate.com/github/codelandev/tribo-viva/compare/pm-payment-gateway-and-shopping-cart :broken_heart:

ssjr commented on config/routes.rb:
```diff
@@ -1,20 +1,37 @@
 
+  get 'purchases', to: 'purchases#index', as: :purchases
+  # checkout/update is used by Iugu to trigger payment update callbacks
+  post 'purchases/update', to: 'purchases#update', as: :purchase_update
```
isso era o que esqueci de comentar:
-> tu tem um código open-source
-> o usuário sabe disso
-> o usuário sabe o invoice_id pq ele recebe no email
-> o usuário é um desenvolvedor e manja dos paranauê
-> ele vê a documentação do iugu sobre notificações
-> ele faz um post manualmente na aplicação dizendo que o status tá pago
-> sistema interpreta como tudo ok e muda o status pra pago
-> usuário vai e coleta os produtos
-> final do mês não vai fechar as contas
-> nos ferramos
o que eu faria nesse caso?

``` ruby
# config/routes.rb
post "/purchases/#{Rails.application.secrets.iugu_url}/update", to: 'purchases#update', as: :purchase_update
```

``` yaml
# config/secrets.yml
development:
  iugu_url: some_random_string
production:
  iugu_url: <% ENV['SOME_ENV_VAR'] %>
```

e no painel do iugu alterar a url do gatilho pra ter o valor de `SOME_ENV_VAR`

patrickemuller replied:
Can we discuss this a little more before implement some fix?

patrickemuller committed:
Organizes gemfile and put fixes version on Iugu gem

patrickemuller committed:
Removes unused order.rb from admin

patrickemuller committed:
Moves iugu key & env to a #widget class inside HTML

patrickemuller committed:
Removes console.log from checkout.js.coffee

patrickemuller committed:
Removes unused `except: :update` on checkouts_controller.rb

patrickemuller committed:
Improves params variable on checkouts_controller.rb

patrickemuller committed:
Removes unused order variable from checkouts_controller.rb

patrickemuller committed:
Removes protect from forgery & adds 404 when success not found

patrickemuller committed:
Just 'store_location' if request is a GET

patrickemuller committed:
Improves code readability on purchases_controller.rb

patrickemuller committed:
Moves store_location method to application_controller.rb

patrickemuller committed:
CHanges validations for order & purchase models

patrickemuller committed:
Fixes irreversible migration for add_devise_to_users

patrickemuller committed:
Fixes problems with wrong spell of checkouts controller spec

patrickemuller committed:
Comments unused method `total` on purchase model

ssjr commented on app/views/checkouts/checkout.html.slim:
```diff
@@ -0,0 +1,75 @@
+                      span.btn.btn--md = number_to_currency offer.value*item['quantity']
+
+            .purchase-form
+              #iugu-widget[data-account="#{Rails.application.secrets.iugu_account_id}" data-environment="#{Rails.env}"]
```
```
Rails.application.secrets.iugu_account_id
=> nil
```

ssjr committed:
Move Cart logic to a separate class

patrickemuller committed:
Fixes validation for purchase button on offers#show

patrickemuller committed:
Removes unused json responses on offers_controller.rb

patrickemuller committed:
Adds iugu_auth_code to authenticate iugu webhooks

patrickemuller committed:
Adds missing iugu_account_id to secrets.yml

ssjr commented on config/routes.rb:
```diff
@@ -1,20 +1,37 @@
 
+  get 'purchases', to: 'purchases#index', as: :purchases
+  # checkout/update is used by Iugu to trigger payment update callbacks
+  post 'purchases/#{Rails.application.secrets.iugu_auth_code}/update', to: 'purchases#update', as: :purchase_update
```
double `"`

gabeborges committed:
Changes have_stock method name and fix it

gabeborges committed:
Adds style to header shopping cart

gabeborges committed:
Creates devise views

gabeborges committed:
Adds style to devise pages

gabeborges committed:
Adds style to cart page

gabeborges committed:
Improves user model and admin

gabeborges committed:
Adds offer address hint

gabeborges committed:
Adds style to checkout page

gabeborges committed:
Improves checkout page style

gabeborges committed:
Changes success text

gabeborges committed:
Improves cart table responsiviness

gabeborges committed:
Adds remove cart item button style

gabeborges committed:
Adds transfer method payment button to checkout page

gabeborges committed:
Creates terms page and adds style to it

gabeborges committed:
Creates finished offers page

gabeborges committed:
Adds social icons to about section in home

gabeborges committed:
Adds mailchimp to about section in homepage

patrickemuller committed:
Fixes problems with registrations#new page

patrickemuller committed:
Adds missing information to registrations#edit page

Also fixes wrong class for phone/cpf for offers#new_purchase page

patrickemuller committed:
Fixes problems with alert messages after sign_up & sign_in

patrickemuller committed:
Fixes wrong value of offer on checkouts_controller.rb

patrickemuller committed:
Fixes is_valid_offer? method on Offer Model

Also fixes IFs on offers#show & pages#home

patrickemuller committed:
Fixes problems with admin/offer & admin/old_purchase pages

patrickemuller committed:
Adds action to remove item from cart

patrickemuller committed:
Adds interaction for transactions costs on checkouts#checkout page

patrickemuller committed:
Adds taxes to iugu charge process on back-end

gabeborges committed:
Improves terms text

gabeborges committed:
Adds buttons to homepage about section

patrickemuller committed:
Block bank_slip when business day is low than 2 days before offer ends

ssjr committed:
Add old method of transfer to Purchases

ssjr committed:
FIX missed JS Formatter file

patrickemuller committed:
Changes text for terms page

patrickemuller committed:
Fixes mailers, success pages, typo for terms & more...

gabeborges committed:
Improves terms text

gabeborges committed:
Improves layout

patrickemuller committed:
Adds Dotenv to Gemfile

patrickemuller committed:
Moves old purchase flow to inside new purchase flow

This includes the views, the old purchase flow, the bank transfer payment method and a lot more.

patrickemuller committed:
Fixes process_payment & values on purchase mailers

patrickemuller committed:
Improves checkouts#success & admin/purchases

patrickemuller committed:
Removes unnecessary files from old purchases

patrickemuller committed:
Translate routes

patrickemuller committed:
Improves homepage messages & organize meta OGs

patrickemuller committed:
Changes newrelic.yml to work on staging

patrickemuller committed:
Removes unused gems from Gemfile

patrickemuller committed:
Fixes problems with CSRF token when IUGU update purchases

patrickemuller committed:
Adds asset_sync gem to Gemfile

This gem allows the assets to be served from S3 instead of Heroku application

gabeborges committed:
Improves mailers and success page

gabeborges committed:
Improves old offer page

gabeborges committed:
Improves about page text

patrickemuller committed:
Adds taxes amount to items of charge on checkouts controller

patrickemuller committed:
Removes duplicated messa after charge success on checkouts controller
